src/EditorSidebar.tsx

```diff
@@ -95,8 +96,8 @@ export default function EditorSidebar() {
                       key={fields[index].id}
                       removeComponents={handleRemoveComponent}
                       index={index}
-                      component={component}
                       componentTitle={componentTitle}
+                      componentId={id}
```

Comment:
Does the component render here before getting the `id`? Does it cause any problems?